File: src/match-wl.pl
#!/usr/bin/perl -w
# -*- coding: utf-8 -*- 

$wln1=shift;
$wln2=shift;
#$outn=shift;

($wl1)=split(' ',`wc -l $wln1`);
($wl2)=split(' ',`wc -l $wln2`);
$top=min($wl1,$wl2);
$cmd=sprintf('paste %s %s | head -%d ',$wln1, $wln2, $top);
print `$cmd`;
if ($wl1>$wl2) {
    open(F,$wln2);
    $w2=<F>;
    close(F);
    open(F,sprintf("tail -n +%d %s |",$top+1,$wln1));
    while (<F>) {
        chomp;
        print "$_\t$w2";
    }
}
sub min{
    ($_[0]<$_[1]) ? $_[0] : $_[1]
}

File: src/match-wl.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*- 

import sys, os
wln1,wln2,outn=sys.argv[1:]
wl1=int(os.popen('wc -l '+wln1).readline().split()[0])
wl2=int(os.popen('wc -l '+wln2).readline().split()[0])
print('paste %s %s | head -%d >%s' % (wln1, wln2, min(wl1,wl2), outn))
os.system('paste %s %s | head -%d >%s' % (wln1, wln2, min(wl1,wl2), outn))
if wl1>wl2:
    with open(wln2) as f:
        w2=f.readline().strip() # just the first word
    outf=open(outn,'a')
    with open(wln1) as f:
        for l in f:
            print(l.strip()+'\t'+w2,file=outf)
